config/config.go

```diff
@@ -109,18 +109,29 @@ func (cfg *Config) NewNode(ctx context.Context) (host.Host, error) {
 		swrm.Filters = cfg.Filters
 	}
 
-	var h host.Host
-	h, err = bhost.NewHost(ctx, swrm, &bhost.HostOpts{
+	h, err := bhost.NewHost(ctx, swrm, &bhost.HostOpts{
 		ConnManager:  cfg.ConnManager,
 		AddrsFactory: cfg.AddrsFactory,
 		NATManager:   cfg.NATManager,
 		EnablePing:   !cfg.DisablePing,
 	})
+
 	if err != nil {
 		swrm.Close()
 		return nil, err
 	}
 
+	if cfg.Relay {
+		// If we've enabled the relay, we should filter out relay
+		// addresses by default.
+		//
+		// TODO: We shouldn't be doing this here.
+		oldFactory := h.AddrsFactory
+		h.AddrsFactory = func(addrs []ma.Multiaddr) []ma.Multiaddr {
+			return relay.Filter(oldFactory(addrs))
+		}
+	}
```

Comment:
This was the real issue. Ideally we'd inject this from the relay code but we can't assert that we have a basic host there.

Reply:
Maybe we should reverse the order of these two filters here. Perhaps the user address factory wants to add some (specific) relay addrs to advertise.